### PR TITLE
Use ethereum and local signer from keep-common

### PR DESCRIPTION
**Pull request by pdyraga**
~~Depends on https://github.com/keep-network/keep-common/pull/61~~

`EthereumSigner` provides functions to sign a message and verify a
signature using Ethereum-specific signature format. It also provides
functions for the conversion of a public key to address.

So far, this code existed only in `keep-core` and could not be reused
from `keep-ecdsa`. In #61 the code has been extracted from `keep-core` 
with slight modifications that will allow it to be imported both from 
`keep-core` and `keep-ecdsa`.

The code is being extracted to allow us to decouple ethereum-specific
implementation functions we use in generic code of `keep-ecdsa` client,
especially in `node.go`.

In #61 also the `local` chain implementation of `keep-core`'s `localSigner`
has been extracted to `keep-common`. This way, we'll reuse the same
code from `keep-core` and `keep-ecdsa` instead of duplicating it.
Here, we remove the `keep-core` implementation and import the one from
`keep-ecdsa`.

### CODE REVIEW

File: pkg/chain/ethereum/ethereum.go [hunk 1]
```diff
@@ -22,15 +22,19 @@ import (
...
+func (ec *ethereumChain) ThresholdRelay() relayChain.Interface {
 	return ec
 }
 
 func (ec *ethereumChain) GetKeys() (*operator.PrivateKey, *operator.PublicKey) {
 	return operator.EthereumKeyToOperatorKey(ec.accountKey)
 }
 
-func (ec *ethereumChain) GetConfig() *relaychain.Config {
+func (ec *ethereumChain) Signing() chain.Signing {
```
Comment (nkuba) [inline pkg/chain/ethereum/ethereum.go:33]: Where are we using this function?

Comment (dimpar) [reply pkg/chain/ethereum/ethereum.go:33]: Looks like we use it in `beacon.go` which implements an interface `Signing` from `chain.go`

**Comment by dimpar** [comment]
Once 61 is merged, we'd need to update mod dependencies. I have no more comments here.

File: pkg/chain/ethereum/ethereum.go (hunk 1 above)
Comment (pdyraga) [reply pkg/chain/ethereum/ethereum.go:33]: It's fetched here:
https://github.com/keep-network/keep-core/blob/b6993e6dcf3bfb3c48a68e6e899954c4a220aeef/pkg/beacon/beacon.go#L52

and used later to generate relay entry.

There are also tests:
- `signing_test.go`
- `states_test.go`
- `membership_validator_test.go`
- `dkgtest.go`

**Comment by pdyraga** [comment]
> Once 61 is merged, we'd need to update mod dependencies. I have no more comments here.

We can wait with updating dependencies for another PR. I was thinking about releasing new `keep-commons` version today or later this week.

File: pkg/chain/ethereum/ethereum.go [hunk 2]
```diff
@@ -22,15 +22,19 @@ import (
...
+	return ethutil.NewSigner(ec.accountKey.PrivateKey)
+}
```
Comment (nkuba) [inline pkg/chain/ethereum/ethereum.go:35]: At first I didn't like the naming that `NewSigner` returns `Signing` (but not `Signer`). But `ethutil.NewSigner` returns a struct that implements `chain.Signing`. I think it's fine.

Comment (nkuba) [reply pkg/chain/ethereum/ethereum.go:35]: Nah, it should be `Signer` instead of `Signing`.

**Review by nkuba — APPROVED** [review]
I'm not sure about naming ([comment](https://github.com/keep-network/keep-core/pull/2242/files#r549675122)) but overall the PR looks fine.

**Pull request merged**